ElectronicObserver/Common/Datagrid/ColumnSelectorView.xaml

```diff
@@ -0,0 +1,82 @@
+﻿<common:WindowBase
+	x:Class="ElectronicObserver.Common.Datagrid.ColumnSelectorView"
+	xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	xmlns:common="clr-namespace:ElectronicObserver.Common"
+	xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	xmlns:local="clr-namespace:ElectronicObserver.Common.Datagrid"
+	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	xmlns:ui="http://schemas.modernwpf.com/2019"
+	Title="{Binding ColumnSelector.ColumnsSelection}"
+	Width="800"
+	Height="450"
+	d:DataContext="{d:DesignInstance local:ColumnSelectorViewModel}"
+	ui:WindowHelper.UseModernWindowStyle="True"
+	x:TypeArguments="local:ColumnSelectorViewModel"
+	SizeToContent="WidthAndHeight"
+	mc:Ignorable="d"
+	>
+	<Window.Resources>
+		<local:ColumnSelectorViewModelProxy x:Key="Proxy" DataContext="{Binding}" />
+	</Window.Resources>
+
+	<Grid>
+
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+
+		<DataGrid
+			Grid.Row="0"
+			AutoGenerateColumns="False"
+			CanUserAddRows="False"
+			CanUserDeleteRows="False"
+			ItemsSource="{Binding Columns}"
+			>
+
+			<DataGrid.ColumnHeaderStyle>
+				<Style BasedOn="{StaticResource DefaultDataGridColumnHeaderStyle}" TargetType="{x:Type DataGridColumnHeader}">
+					<Setter Property="Template" Value="{StaticResource DataGridColumnHeaderTemplate}" />
+				</Style>
+			</DataGrid.ColumnHeaderStyle>
+
+			<ui:DataGridHelper.TextColumnElementStyle>
+				<Style BasedOn="{StaticResource DataGridTextBlockStyle}" TargetType="TextBlock">
+					<Setter Property="Padding" Value="5 0" />
+					<Setter Property="FontFamily" Value="{Binding DataContext.Font, Source={StaticResource Proxy}}" />
+					<Setter Property="FontSize" Value="{Binding DataContext.FontSize, Source={StaticResource Proxy}}" />
+					<Setter Property="Foreground" Value="{Binding DataContext.FontBrush, Source={StaticResource Proxy}}" />
+				</Style>
+			</ui:DataGridHelper.TextColumnElementStyle>
+
+			<DataGrid.Columns>
+				<DataGridTextColumn
+					Binding="{Binding Header}"
+					Header="{Binding DataContext.ColumnSelector.Column, Source={StaticResource Proxy}}"
+					IsReadOnly="True"
+					/>
+
+				<DataGridCheckBoxColumn Binding="{Binding Visible}" Header="{Binding DataContext.ColumnSelector.Visible, Source={StaticResource Proxy}}" />
```

Comment:
Looks like DataGridCheckBoxColumn needs 2 clicks to actually get changed, might be better to use DataGridTemplateColumn.

https://stackoverflow.com/questions/3833536/how-to-perform-single-click-checkbox-selection-in-wpf-datagrid
```xml
<DataGridTemplateColumn Header="{Binding DataContext.ColumnSelector.Visible, Source={StaticResource Proxy}}">
	<DataGridTemplateColumn.CellTemplate>
		<DataTemplate>
			<CheckBox
				MinWidth="0"
				Margin="10 0"
				IsChecked="{Binding Visible, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
				/>
		</DataTemplate>
	</DataGridTemplateColumn.CellTemplate>
</DataGridTemplateColumn>
```